codalab/model/bundle_model.py

```diff
@@ -2910,18 +2961,71 @@ def create_bundle_store(self, user, name, storage_type, storage_format, url, aut
         return uuid
 
-    def save_oauth2_auth_code(self, grant):
-        with self.engine.begin() as connection:
-            result = connection.execute(oauth2_auth_code.insert().values(grant.columns))
-            grant.id = result.lastrowid
-        return grant
-
-    def delete_oauth2_auth_code(self, auth_code_id):
+    def update_bundle_store(self, user, uuid, update_fields):
+        """
+        update a bundle store
+        """
         with self.engine.begin() as connection:
             connection.execute(
-                oauth2_auth_code.delete().where(oauth2_auth_code.c.id == auth_code_id)
+                cl_bundle_store.update()
+                .where(and_(cl_bundle_store.c.uuid == uuid, cl_bundle_store.c.owner_id == user,))
+                .values(update_fields)
             )
 
+    def get_bundle_store(self, user, uuid):
+        """
+        Return the bundle store corresponding to the specified uuid.
+        Just returns the owner_id, name, and url of the bundle store.
+        """
+        with self.engine.begin() as connection:
+            row = connection.execute(
+                select(
+                    [
+                        cl_bundle_store.c.uuid,
+                        cl_bundle_store.c.owner_id,
+                        cl_bundle_store.c.name,
+                        cl_bundle_store.c.storage_type,
+                        cl_bundle_store.c.storage_format,
+                        cl_bundle_store.c.url,
+                    ]
+                ).where(
+                    and_(
+                        cl_bundle_store.c.uuid == uuid,
+                        or_(
+                            cl_bundle_store.c.owner_id == 'codalab(0)',
+                            cl_bundle_store.c.owner_id == user,
+                        ),
+                    )
+                )
+            ).fetchone()
+            return {
+                'owner_id': row.owner_id,
+                'name': row.name,
+                'url': row.url,
+                'storage_type': row.storage_type,
+                'storage_format': row.storage_format,
+            }
+
+    def delete_bundle_store(self, user, uuid):
+        """
+        Delete a bundle store given its uuid. We can only delete the Bundle Store if there are
+        no BundleLocations associated with it.
+        """
+        with self.engine.begin() as connection:
+            bundle_store_row = connection.execute(
+                select([cl_bundle_store.c.uuid]).where(
+                    and_(cl_bundle_store.c.uuid == uuid, cl_bundle_store.c.owner_id == user,)
+                )
+            ).fetchone()
+            bundle_location_row = connection.execute(
+                select([cl_bundle_location.c.id]).where(
+                    cl_bundle_location.c.bundle_store_uuid == bundle_store_row.uuid
+                )
+            ).fetchone()
+            # delete only if there is BundleLocation associated
```

Comment:
I'm confused by this comment, I thought we delete bundle stores only if there are no BundleLocations associated with it?